manim/mobject/types/opengl_vectorized_mobject.py

```diff
@@ -89,12 +90,11 @@ def __init__(
         triangulation_locked=False,
         **kwargs,
     ):
-        self.data = {}
-        self.fill_color = fill_color
+        self.data = OpenGLVMobject.data
+        OpenGLVMobject.data = {}
```

Comment:
Is this really required given that you have added `data = {}` just a few lines above as a class attribute?

Reply:
Some classes that inherit from `VMmobject` depend on data being present (because they are setting `fill_color`) before calling `super().__init__(...)`. This class provides it by having the data as class property and once `__init__` is called it swaps the class property with instance property.

Reply:
Hmm. I see, do you have an example for one of the classes that do that? Doesn't sound like intentional design (but then again, it doesn't surprise me and fits the library).

Reply:
`MarkupText` sets color here https://github.com/ManimCommunity/manim/blob/b9587341e0c33b8bf7d179654b88ab6b87253d64/manim/mobject/svg/text_mobject.py#L1036
which in turn sets `fill_color`.

Reply:
I see. Resolving this would require adapting `MarkupText.text2svg` a bit. @naveen521kk, do you have any thoughts about this?

Reply:
I think that assignment isn't required, instead, it can be simply passed as a kwarg for `super().__init__()`.

Reply:
This will break caching, because it uses color and it will no longer be set:
https://github.com/ManimCommunity/manim/blob/b9587341e0c33b8bf7d179654b88ab6b87253d64/manim/mobject/svg/text_mobject.py#L1151

Reply:
Hmm, where does manim store the colour then?

Reply:
This PR intends to change the `color`s to properties so they use a single source of truth: `rgba_array`s, such as `fill_rgbas`.

Reply:
> This PR intends to change the colors to properties so they use a single source of truth: rgba_arrays, such as fill_rgbas.

If I read the changelog
 
> color, stroke_color and fill_color are now properties in VMobject

It says color is an attribute of `VMobject` right and `MarkupText` is in turn a VMobject, why wouldn't the color be set?

Reply:
`MarkupText.text2svg` is called before calling the parent's `__init__` method so `self.color` won't be set at that time if we just pass it to `super().__init__()`.

Reply:
oh, I see, in which I guess it can be set, and later when `super().__init__()` is called it can override it.

Reply:
I've passed the color directly to `text2svg` in order to avoid this problem.